packages/flutter_tools/lib/src/base/analyze_size.dart

```diff
@@ -0,0 +1,336 @@
+import 'dart:convert';
+
+import 'package:flutter_tools/src/base/process.dart';
+import 'package:flutter_tools/src/base/terminal.dart';
+import 'package:intl/intl.dart';
+import 'package:meta/meta.dart';
+import 'package:vm_snapshot_analysis/treemap.dart';
+
+import '../base/file_system.dart';
+import '../globals.dart' as globals;
+
+// This file can be in devtools_shared
+class SizeAnalyzer {
+  static const String kAotSizeJson = 'aot-size.json';
+
+  static String getAotSizeAnalysisJsonFile(String aotOutputPath) {
+    return globals.fs.path.join(aotOutputPath, kAotSizeJson);
+  }
+
+  static String getAotSizeAnalysisExtraGenSnapshotOption(String aotOutputPath) {
+    return '--print-instructions-sizes-to=${getAotSizeAnalysisJsonFile(aotOutputPath)}';
+  }
+
+  // Parse the output of unzip -v which shows the zip's contents' compressed sizes.
```

Comment:
maybe give an example of what the unzip -v output should look like here. That should make it more obvious why this regexp is the right one.